ci/docker/ubuntu-20.04-cpp.dockerfile

```diff
@@ -172,6 +175,7 @@ ENV absl_SOURCE=BUNDLED \
     ARROW_WITH_ZSTD=ON \
     ASAN_SYMBOLIZER_PATH=/usr/lib/llvm-${llvm}/bin/llvm-symbolizer \
     AWSSDK_SOURCE=BUNDLED \
+    AZURE_SDK_SOURCE=BUNDLED \
```

Comment:
Could you use `Azure_SOURCE` instead of `AZURE_SDK_SOURCE` because Azure SDK for C++ uses `Azure` for its CMake package's prefix?